content/blog/fast-data-processing-pipeline-for-predicting-flight-delays-using-apache-apis-kafka-spark-streaming-and-machine-learning-part-3.md

```diff
@@ -0,0 +1,295 @@
+
+Below are some example SQL queries using the Drill shell.
+
+**What is the count of predicted delay/notdelay by origin**
+
+![count of predicted delay/notdelay by origin](/img/count-predicted-delay-notdelay-origin.png)
+
+**What is the count of predicted delay/notdelay by origin and dest?**
+
+![What is the count of predicted delay/notdelay by origin and dest?](/img/count-predicted-delay-notdelay-origin-dest.png)
+
+Follow the instructions in the github code README to [add a secondary index to MapR Database](https://docs.datafabric.hpe.com/62/Drill/optimizing_queries_with_indexes.html) and try more queries using the index.
+
+**Querying with the Open JSON API (OJAI)**
+
+Below is a Java example of using the OJAI Query interface to query documents in a MapR Database JSON table:
+
+![OJAI Query interface to query documents in a MapR Database JSON](/img/ojai-query-interface-mapr-db.png)
+
+Partial output for this query to "find predicted late flights for AA" is shown below:
+
+![find predicted late flights for AA](/img/find-predicted-late-flights-aa.png)
+
+Below are some example OJAI queries using the MapR Database shell.
+
+**What are the SFO to DEN flights that were predicted late ?**
+
+```json
+maprdb> find /apps/flights -where '{"$and":[{"$eq":{"pred_dtree":1.0}},{ "$like" : {"_id":"%SFO_DEN%"} }]}' --f _id,pred_dtree
+```
+
+
+![MapR Database find /apps/flights](/img/mapr-db-find.png)
+
+**Summary**
+
+* In this blog post, you've learned how to consume streaming JSON events, store in a document database, and explore with SQL using Apache Spark, Apache Kafka API, Apache Drill, MapR Event Store, MapR Database, and OJAI.
+
+**Code**
+
+* You can download the code and data to run these examples from here (refer to the README for complete instructions to run): https://github.com/mapr-demos/mapr-es-db-60-spark-flight
+
+**Running the Code**
+
+All of the components of the use case architecture we just discussed can run on the same cluster with the MapR Data Platform. The MapR Data Platform integrates global event streaming, real-time database capabilities, and scalable enterprise storage with a collection of data processing and analytical engines to power data processing pipelines and intelligent applications.
+
+![MapR Data Platform](/img/mapr-cdp.png)
+
+This example was developed using the [MapR 6.0 container for developers](https://docs.datafabric.hpe.com/62/MapRContainerDevelopers/MapRContainerDevelopersOverview.html), a docker container that enables you to create a single node MapR cluster. The container is lightweight and designed to run on your laptop. (refer to the code README for instructions on running the code).
+
+You can also look at the following examples:
+
+* [MapR Database 60-getting-started](https://github.com/mapr-demos/mapr-db-60-getting-started) to learn Discover how to use DB Shell, Drill and OJAI to query and update documents, but also how to use indexes.
```

Comment:
to learn Discover --> suggest to rephrase: "to discover how to use DB Shell,..."

Reply:
done